src/main/java/com/amazon/ion/impl/IonReaderContinuableCoreBinary.java

```diff
@@ -1248,14 +1245,30 @@ private boolean valueUnavailable() {
             return event == Event.NEEDS_DATA || event == Event.NEEDS_INSTRUCTION;
         }
 
-        private void classifySexpWithinEncodingDirective() {
+        private void classifyDirective() {
+            errorIf(getEncodingType() != IonType.SYMBOL, () -> "Ion encoding directives must start with a directive keyword.");
```

Comment:
Are these anonymous lambdas free? Should we just pass in a string literal, since that appears to be how `errorIf` is always used?

Reply:
I don't think they're free. This is a holdover from porting `errorIf` from the equivalent thing I wrote for `EncodingDirectiveReader` which is written in Kotlin where it is free to use certain functional parameters for `inline` functions.